### PR TITLE
Reproduce expert-concept policies on a simulated MARL task

### LINKED ISSUE #59: [Notebook] Reproduce expert-concept policies on one simulated MARL task
Parent roadmap: #60  
Library dependencies: #52 and #53

## Paper

[Concept Learning for Interpretable Multi-Agent Reinforcement Learning](https://proceedings.mlr.press/v205/zabounidis23a.html)

## Outcome

Deliver `docs/source/reproductions/marl-concept-policy.ipynb`, a dedicated scientific reproduction notebook separate from tutorials.

For one declared simulated cooperative-competitive task, the notebook should:

1. train a policy that predicts expert concepts and consumes them for decisions;
2. compare concept and policy metrics with a matched non-concept policy;
3. intervene on concept predictions through XDRL's existing typed TensorDict intervention boundary; and
4. test whether corrections change behaviour/performance in the predicted direction.

## Notebook requirements

- Pin the paper/supplement, task implementation, concept definitions, architecture, training config, seeds, and evaluation split.
- Record whether original code/checkpoints exist; enumerate every underspecified reimplementation choice.
- Match parameter, data, optimizer, and evaluation budgets for concept/non-concept controls where the source permits.
- Preserve per-agent concept/action axes and use explicit named reductions.
- Report held-out concept metrics, policy metrics, stability/sample-efficiency summaries, intervention effects, and uncertainty separately.
- Record external inputs and full-result artifacts through #53.
- Include a bounded smoke path clearly marked as non-scientific.

## Acceptance criteria

- [ ] The dedicated notebook renders from the Reproductions gallery introduced by #52.
- [ ] Correct, incorrect, no-op, and/or shuffled concept interventions use a frozen evaluation set.
- [ ] Existing public XDRL multi-agent and TensorDict intervention APIs are used; any newly discovered library gap becomes a focused design/feature issue.
- [ ] Reimplementation deviations and unavailable reference artifacts are explicit.
- [ ] Claims remain limited to the simulated task; the real-world result is outside scope.

## Non-goals

This issue does not reproduce the real-robot experiment or add a MARL trainer to XDRL.

### PR DESCRIPTION
**Pull request by Xmaster6y**
## Summary

- add a dedicated expert-concept MARL reproduction notebook to the Reproductions gallery
- pin the paper contract and fail closed when exact FortAttack code, checkpoints, splits, and reference outputs are unavailable
- train matched concept/non-concept smoke controls across five seeds with explicit per-agent and concept-group reductions
- run correct, incorrect, shuffled, and no-op concept interventions through the public TensorDict/TDHook boundary with artifact provenance and paired bootstrap uncertainty

## Evidence boundary

The executable path is a clearly labeled synthetic 2v2 lane-defense smoke fixture. It validates training, analysis, intervention, and provenance mechanics only; it does not reproduce FortAttack, MAPPO results, or the real-robot experiment.

## Validation

- `uv run jupyter nbconvert --to notebook --execute docs/source/reproductions/marl-concept-policy.ipynb ...`
- paper-mode fail-closed check
- `uv run pre-commit run --all-files`
- `uv run pytest -q` (284 passed)
- `uv run sphinx-build -W --keep-going -b html docs/source build/docs/html`

Closes #59

### CODE REVIEW

File: docs/source/reproductions/marl-concept-policy.ipynb [hunk 1]
```diff
@@ -0,0 +1,1283 @@
...
+    "        \"all_agents_correct_by_episode\": correct.all(dim=1).float(),\n",
+    "        \"action_by_episode_agent\": action_logits.argmax(-1),\n",
+    "    }\n",
+    "\n",
+    "\n",
+    "def bootstrap_interval(values, *, seed, draws=500):\n",
+    "    generator = torch.Generator().manual_seed(seed)\n",
+    "    estimates = []\n",
+    "    for _ in range(draws):\n",
+    "        indices = torch.randint(len(values), (len(values),), generator=generator)\n",
```
[the remark this answers is not in the document]
Comment (Xmaster6y) [reply docs/source/reproductions/marl-concept-policy.ipynb]: Addressed in 8057d7b. All correct, incorrect, and shuffled interventions now run against each of the five independently trained models on the same frozen 96-episode TensorDict (15 paired workflows total). The notebook retains explicitly labeled within-seed episode intervals, reduces each model to one paired mean effect, and reports the primary bootstrap over those five seed-level means with `n_training_seeds: 5` and seed IDs. The directional gate now uses that training-seed interval.

File: docs/source/reproductions/marl-concept-policy.ipynb [hunk 2]
```diff
@@ -0,0 +1,1283 @@
...
+    "            \"frozen\": True,\n",
+    "            \"ordered_batch_axes\": [\"episode\", \"agent\"],\n",
+    "            \"episodes\": EVALUATION_EPISODES,\n",
+    "            \"agents\": AGENTS,\n",
+    "            \"components\": [name for name, _ in frozen_items],\n",
+    "        },\n",
+    "    ),\n",
+    "    InputArtifactReference(\n",
+    "        \"reference:zabounidis-et-al-2023-paper\",\n",
+    "        InputArtifactRole.REFERENCE_RESULT,\n",
```
[the remark this answers is not in the document]
Comment (Xmaster6y) [reply docs/source/reproductions/marl-concept-policy.ipynb]: Addressed in 8057d7b. The paper PDF is now `InputArtifactRole.OTHER`, matching the supplement, with metadata identifying it as a document and recording `reference_results_available: False`. No artifact is classified as `REFERENCE_RESULT` while reference outputs remain unavailable.
